### ISSUE TITLE
Question: what happened to codecommit?

### ISSUE THREAD

**Issue by slimbentami** [paws-r/paws#375]
I would like to just list the content of an aws codecommit repository from R code. I can't seem to find anything that works. is there anything in paws that could achieve this? 
Thank you

this [https://paws-r.github.io/docs/codecommit/](url) looked the answer but then it does not seem to exist anymore ....

**Comment by adambanker** [comment]
Thanks for submitting this issue! CodeCommit is not currently available from CRAN, but you can install it directly from this repository with the following command: `devtools::install_github("paws-r/paws/cran/paws.developer.tools")`.

You can then use codecommit as follows:
```r
## To initialize a service object
codecommit <- paws.developer.tools::codecommit()

## To view documentation
?paws.developer.tools::codecommit
```

It should be fully functional, but I have not used CodeCommit extensively, so it is possible there are bugs.

**Comment by slimbentami** [comment]
Thank you! 
When I run the install_github, I get the following message:
`Downloading GitHub repo paws-r/paws@master
Error in utils::download.file(url, path, method = method, quiet = quiet,  : 
  cannot open URL 'https://api.github.com/repos/paws-r/paws/tarball/master'`

**Comment by davidkretch** [comment]
Seems like like it could be a network related issue. Are you able to download other files, e.g. `download.file("https://github.com", "test.html")`?

**Comment by slimbentami** [comment]
Thank you, I am able to download files, the example you provided worked fine. I am still getting the same message on the install_github command however.

**Comment by davidkretch** [comment]
There was [an issue in the `remotes` repo](https://github.com/r-lib/remotes/issues/130) about this problem. They state that the file download problem is likely due to a rate limit or some other problem, like bad proxy setup.

You may be able to get around it by changing the `download.file` download method with `options(download.file.method = "libcurl")`.

Failing that you can manually [download the repo](https://github.com/paws-r/paws/archive/main.zip) and install it locally with `devtools::install_local("paws/cran/paws.developer.tools")`, assuming your working directory is the parent folder of the unzipped file.

**Comment by slimbentami** [comment]
Thank you! the manual download worked! Now on to interacting with codecommit .....

**Comment by davidkretch** [comment]
Great. I have also not used this API, but my guess is that the [`get_folder` API operation](https://paws-r.github.io/docs/codecommit/get_folder.html) will get you a list of a repository's contents:

```
codecommit <- paws.developer.tools::codecommit()
codecommit$get_folder(
  repositoryName = "string",
  commitSpecifier = "string",
  folderPath = "string"
)
```

**Comment by davidkretch** [comment]
Closing for now.

**Comment by davidkretch** [comment]
FYI CodeCommit is now available in the version of paws on CRAN (0.1.12).
